### PR TITLE
Gerrit Fritz Aufgabe 6 (1778099, 1623571)

### PR DESCRIPTION
**Pull request by GeFritz**

### CODE REVIEW

File: Exercises/GerritFritz/exercises_06.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#Gerrit Fritz
+#15.11.2023
+
+from __future__ import annotations
+
+class Directory():
```
Comment (ClasherKasten) [inline Exercises/GerritFritz/exercises_06.py:6]: Python 3 doesn't require the parentheses after the class name.

File: Exercises/GerritFritz/exercises_06.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            
+    def update_size(self, size: int) -> None:
+        self.size += size
+        if self.parent != None: self.parent.update_size(size) 
+    
+    def get_dir(self, folder: str) -> Directory:
+        if folder in self.folders.keys(): return self.folders[folder]
+        elif folder == "..": return self.parent
+        elif folder == "/": return self.root
+        else: raise RuntimeError(f"No such dir: {folder}")
```
Comment (ClasherKasten) [inline Exercises/GerritFritz/exercises_06.py:30]: Please refrain from writing blocks like that. It's not very pythonic to do it, especially not to such an extent.

File: Exercises/GerritFritz/exercises_06.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+def get_mindir(size_dict:dict, max:int, free:int)->int: 
+    return min([val for val in size_dict.values() if val >= size_dict["/"]+free-max])
+
+    
+
+root = build_filesys(get_record("data//terminal_record.txt"))
+size_dict = root.get_dict()
+print(get_sum(size_dict))
+print(get_mindir(size_dict,70000000, 30000000))
```
Comment (ClasherKasten) [inline Exercises/GerritFritz/exercises_06.py:70]: No magic values!

**Review by ClasherKasten — CHANGES_REQUESTED** [review]
Almost good type annotations but types inheriting from `Generic` should have concrete types.

File: Exercises/GerritFritz/exercises_06.py (hunk 3 above)
Comment (GeFritz) [reply Exercises/GerritFritz/exercises_06.py:70]: Could you please elaborate?

File: Exercises/GerritFritz/exercises_06.py (hunk 2 above)
Comment (GeFritz) [reply Exercises/GerritFritz/exercises_06.py:30]: What exactly is the problem here? Is not merely a preference?

File: Exercises/GerritFritz/exercises_06.py (hunk 3 above)
Comment (ClasherKasten) [reply Exercises/GerritFritz/exercises_06.py:70]: Someone without access to the assignment text shouldn't have to guess why exactly these values are chosen.

**Comment by GeFritz** [comment]
Thank you for the insights. The change requests are helpful, but normally the classmates have to review and get grades. The problem now is that the others have to review code that already was reviewed. I never saw a pull request from you, so I assume that you aren't in our course?

File: Exercises/GerritFritz/exercises_06.py (hunk 2 above)
Comment (ClasherKasten) [reply Exercises/GerritFritz/exercises_06.py:30]: You probably won't stumble a lot about productive code like that. Which in the end means for you to adapt to the new (way more common and standardized) pattern.

See [PEP8](https://peps.python.org/pep-0008/#other-recommendations)
> Compound statements (multiple statements on the same line) are generally discouraged

**Comment by ClasherKasten** [comment]
> The problem now is that the others have to review code that already was reviewed.

Not really a problem. Your course can have special standards which could invalidate/complement the review.

**Comment by LeMixer** [comment]
> > The problem now is that the others have to review code that already was reviewed.
> 
> Not really a problem. Your course can have special standards which could invalidate/complement the review.
> 
> 

Yes, it is a problem. The weekly reviews a part of our graded weekly tasks. This repo is meant to be a platform for our class to practice writing and reviewing Python code. Therefore it's not your business with whatever argumentation to leave reviews potentially disallowing us from writing a thorough review on this pull request. So please refrain from writing reviews in this repo in the future!

**Comment by GeFritz** [comment]
> Almost good type annotations but types inheriting from `Generic` should have concrete types.

what exactly should i change?

**Review by Reisii — APPROVED** [review]
Nice work. Code works as intended and nice use of classes. Imo type annotations aren't necessary, so no reason for it to be a reason to request changes.

**Pull request merged**